sql/analyzer/resolve_columns.go

```diff
@@ -246,12 +246,61 @@ func qualifyColumns(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, sel
 		symbols = getNodeAvailableNames(n, scope, symbols, nestingLevel)
 		nestingLevel++
 
-		return transform.OneNodeExprsWithNode(n, func(n sql.Node, e sql.Expression) (sql.Expression, transform.TreeIdentity, error) {
+		// Updates need to have check constraints qualified since multiple tables could be involved
+		checkConstraintsUpdated := false
+		if nn, ok := n.(*plan.Update); ok {
+			newChecks, err := qualifyCheckConstraints(nn)
+			if err != nil {
+				return n, transform.SameTree, err
+			}
+			nn.Checks = newChecks
+			n = nn
+			checkConstraintsUpdated = true
+		}
+
+		newNode, identity, err := transform.OneNodeExprsWithNode(n, func(n sql.Node, e sql.Expression) (sql.Expression, transform.TreeIdentity, error) {
 			return qualifyExpression(e, symbols)
 		})
+		if err != nil {
+			return n, transform.SameTree, err
+		}
+		if checkConstraintsUpdated {
+			identity = transform.NewTree
+		}
+		return newNode, identity, nil
 	})
 }
 
+// qualifyCheckConstraints returns a new set of CheckConstraints created by taking the specified Update node's checks
+// and qualifying them to the table involved in the update, including honoring any table aliases specified.
+func qualifyCheckConstraints(update *plan.Update) (sql.CheckConstraints, error) {
+	checks := update.Checks
+	table, alias := getResolvedTableAndAlias(update.Child)
+
+	newExprs := make([]sql.Expression, len(checks))
+	for i, checkExpr := range checks.ToExpressions() {
+		newExpr, _, err := transform.Expr(checkExpr, func(e sql.Expression) (sql.Expression, transform.TreeIdentity, error) {
+			switch ee := e.(type) {
```

Comment:
3 nits: 1) zach recommended that i should block the parent definition in a switch to narrow my reference variables when possible (`switch e := e.(type)`); 2) maybe use `UnresolvedColumn` here, not sure we are interested in `GetField` (also i didn't know we had the `column` interface); 3) we usually add a `default` for every switch statement, even if empty.